astronomer/providers/google/cloud/operators/dataproc.py

```diff
@@ -190,6 +190,14 @@ def execute(self, context: Context) -> None:
             metadata=self.metadata,
         )
 
+        try:
+            hook.get_cluster(project_id=self.project_id, region=self.region, cluster_name=self.cluster_name)
+        except NotFound:
+            self.log.info("Cluster deleted.")
```

Comment:
I would suggest that we log here that `Cluster not found. Skipping deletion.` instead of saying cluster deleted.

Reply:
I'd said this might make the log less accurate as we're still deleting the cluster [here](https://github.com/astronomer/astronomer-providers/blob/24f2b6100ea0ee5a777f849c0aada19adcaf960a/astronomer/providers/google/cloud/operators/dataproc.py#L183). WDYT

Reply:
I'll merge it for now. But we can send another PR if we decide to change the log 🙂

Reply:
ohh, yes, sorry, you're right. In my mind, I thought we're checking for the cluster status before triggering the delete.